requirements-docs.txt

```diff
@@ -5,6 +5,7 @@
 # build
 docutils >=0.18
 jupyter-server-mathjax
+jupyterlab >=3.5.3,<3.6
```

Comment:
already handled by `-> test -> build`? More lab specs in more places is harder to reason about...

Reply:
also all these `requirements-*` and `environment.yml` are quite confusing. Hoping we can do something about it as part of #684.

Reply:
> all these requirements-* and environment.yml

At the expense of more downloads in CI, we could presumably go down to:

- `requirements.txt`
- `.binder/environment.yml`
  - really only `diffoscope` is not available on all platforms

It really would be nice to have slightly more control over the `pip` cache behavior than the current setup, as it doesn't seem to clean up very efficiently, so these just keep getting bigger.

> do something 

i sure hope the _something_ is not "use hatch for everything," as shoehorning a single-package build tool into a monorepo manager is not very enticing. Similarly, i continue to be unimpressed with `nx`... maybe it's awesome if for teams that use all the distributed cache stuff, but mostly just seems to rework stuff a lot, and get confused about its cache state.

Reply:
> "use hatch for everything,"

Not really no. We don't use `hatch` in a fancy way here, and we are also not using `hatch-jupyter-builder` (and not planning to). Just so that's it the same tool as in other Jupyter projects.

Probably more to be discussed in https://github.com/jupyterlite/jupyterlite/issues/684 and follow-ups, but I was more thinking about leveraging top-level scripts for building packages and run tasks. So folks could contribute without having to know `doit` or understand / debug the whole `dodo.py`.